### PR TITLE
fix deprecations and remove obsolete extensions and protocols

### PR DESCRIPTION
**Pull request by petropavel13**

### CODE REVIEW

File: Sources/Extensions/Array/Array+Extensions.swift [hunk 1]
```diff
@@ -20,52 +20,10 @@
...
-                    intersection.append(item)
-                }
-            }
-        }
-
-        return intersection
-    }
-
-    /// Find unique elements in array compared to other arrays
-    func subtract(values: [Array.Element]...) -> Array {
```
Comment (scoreyou) [inline Sources/Extensions/Array/Array+Extensions.swift]: Disagree with removing this extension

File: Sources/Extensions/UIAlertController/UIAlertController+Extensions.swift [hunk 2]
```diff
@@ -1,47 +0,0 @@
...
-
-    /**
-     Creates UIAlertController instance
-     
-     - parameter item: information about alert in AlertRepresentable form
-     - parameter actions: actions handlers
-     
-     - returns: UIAlertController instance
-     */
-    static func alert(item: AlertRepresentable, actions: (() -> Void)?...) -> UIAlertController {
```
Comment (scoreyou) [inline Sources/Extensions/UIAlertController/UIAlertController+Extensions.swift]: disagree

File: Sources/Extensions/UIDevice/UIDevice+Extensions.swift [hunk 3]
```diff
@@ -1,32 +0,0 @@
...
-//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-//  THE SOFTWARE.
-//
-
-import UIKit
-
-public extension UIDevice {
-
-    /// Returns true if the current device is simulator
-    static var isSimulator: Bool {
```
Comment (scoreyou) [inline Sources/Extensions/UIDevice/UIDevice+Extensions.swift]: hhmm, why do you delete this?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,18 @@
 # Changelog
 
 ## 0.6.0
-- **New**: Swift 4 support & dependencies update
+- **New**: Swift 4 support & dependencies update.
+- **Remove**: `Mutex`
+- **Remove**: `IndexPath+ImmutableIndexPath`
+- **Remove**: `StoryboardProtocol`, `StoryboardProtocol+Extensions`, `StoryboardProtocol+DefaultBundle`
+- **Remove**: `String+Extensions`
```
Comment (scoreyou) [inline CHANGELOG.md]: just image part here

File: Sources/Extensions/UIDevice/UIDevice+Extensions.swift (hunk 3 above)
Comment (petropavel13) [reply Sources/Extensions/UIDevice/UIDevice+Extensions.swift]: use library. https://github.com/schickling/Device.swift/blob/master/Device/UIDeviceExtension.swift#L74

Comment (scoreyou) [reply Sources/Extensions/UIDevice/UIDevice+Extensions.swift]: ok here

File: Sources/Extensions/UIView/UIView+LoadFromNib.swift [hunk 5]
```diff
@@ -30,7 +30,7 @@ public extension UIView {
     /// - Parameter bundle: The bundle in which to search for the nib file.
     ///                     If you specify nil, this method looks for the nib file in the main bundle.
     /// - Returns: UIView or UIView subclass instance
-    static func loadFromNib<T>(bundle: Bundle? = nil) -> T where T: XibNameProtocol, T: UIView {
+    static func loadFromNib<T>(bundle: Bundle? = nil) -> T where T: UIView {
```
Comment (Ipomoea) [inline Sources/Extensions/UIView/UIView+LoadFromNib.swift]: did you check this function?

**Pull request merged**